### PR TITLE
feat: add balance property to NftItem

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds an optional `balance` property for ERC-1155 `NftItem`s, populated from Alchemy, which is our only reliable source as far as token standards are concerned. Note, since we don't upsert to portfolio/assets slice yes, this doesn't do anything yet.

When this is merged, 4712 will need a restack and we'll need to add the proper balance for 1155s to

https://github.com/shapeshift/web/pull/4712/files#diff-7ef9f6fd70f7be4faed5cc8c2702f7d46529689cf81962a71d88be8c46a23025R167

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Relates to https://github.com/shapeshift/web/issues/4703 - `patches sends (accurate portfolio balances from nft api sources)`

## Risk

None, not consumed yet

## Testing

- None

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4710** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4710" target="_blank">Graphite</a> 
    * **PR #4722** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4722" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4722?utm_source=stack-comment).

File: src/state/apis/nft/parsers/alchemy.ts [hunk 1]
```diff
@@ -159,13 +159,20 @@ export const parseAlchemyNftToNftItem = async (
 
   const medias = maybeMedias.unwrap()
 
+  const maybeBalance: Pick<NftItem, 'balance'> = (
+    alchemyNft.tokenType.toLowerCase() === ASSET_NAMESPACE.erc1155
+      ? { balance: bnOrZero((alchemyNft as OwnedNft).balance).toNumber() }
+      : {}
+  ) as Pick<NftItem, 'balance'>
```
Comment (0xApotheosis) [inline src/state/apis/nft/parsers/alchemy.ts:166]: Any reason this isn't the below?

```ts
const maybeBalance: Pick<NftItem, 'balance'> =
    alchemyNft.tokenType.toLowerCase() === ASSET_NAMESPACE.erc1155
      ? { balance: bnOrZero((alchemyNft as OwnedNft).balance).toString() }
      : {}
```

Comment (gomesalexandre) [reply src/state/apis/nft/parsers/alchemy.ts]: Disregard me I'm derp - bd8cac55a2fdc5cfbcfef844bcdd62ce7b20d0c2

**Review by 0xApotheosis — APPROVED** [review]
gm

**Pull request merged**